sycl/source/detail/kernel_impl.hpp

```diff
@@ -383,6 +416,90 @@ inline typename syclex::info::kernel_queue_specific::max_num_work_groups::
                                DynamicLocalMemorySize);
 }
 
+template <>
+inline typename syclex::info::kernel_queue_specific::max_work_group_size::
+    return_type
+    kernel_impl::ext_oneapi_get_info<
+        syclex::info::kernel_queue_specific::max_work_group_size>(
+        queue Queue) const {
+  const auto &Adapter = getAdapter();
+  const auto DeviceNativeHandle =
+      getSyclObjImpl(Queue.get_device())->getHandleRef();
+
+  size_t KernelWGSize = 0;
+  Adapter->call<UrApiKind::urKernelGetGroupInfo>(
+      MKernel, DeviceNativeHandle, UR_KERNEL_GROUP_INFO_WORK_GROUP_SIZE,
+      sizeof(size_t), &KernelWGSize, nullptr);
+  return KernelWGSize;
+}
+
+template <int D>
+inline sycl::id<D> generate_id(const sycl::range<D> &DevMAxWorkItemSizes,
+                               const size_t DevWgSize) {
+  sycl::id<D> ret;
+  for (int i = 0; i < D; i++) {
```

Comment:
```suggestion
  sycl::id<D> Ret;
  for (int I = 0; I < D; I++) {
```

However, I'm not sure if RT code fully adheres to LLVM Coding Standards, will be good to hear some feedback from RT reviewers group.